src/shared.ts

```diff
@@ -132,7 +133,7 @@ export function getIndexByLetter(
 	startIndex = 0
 ): number {
 	const orderedOptions = [...options.slice(startIndex), ...options.slice(0, startIndex)]
-	const firstMatch = filterOptions(orderedOptions, filter)[0]
+	const firstMatch = filterOptions(orderedOptions, filter, this.labelField)[0]
```

Comment:
So, one thing I'll want to include w/ this PR is an update to the `SelectOption` interface. It obviously needs to be way more permissive now. 

Something like
```ts
interface SelectOption {
  label?: string
  value?: string
  [key: string]: string
}
```

^ that would be great, except we have one more consideration: we want people to be able to track other things on their options if they want to. Namely, if I have a collection of items for a select, and the actual value for each item that I'm interested in tracking is another object, or an array, I should be OK to include that. 

This PR would be fine with that so far, *so long as* the fields chosen for `labelField` and `valueField` are strings, *and* that the `valueField` is unique for each option.

Those seem like reasonable requirements, that will just require a lot of documentation & possibly some JS checks. It's hard to document that in TypeScript, so it's almost that the best we can do for `SelectOption` is
```ts
interface SelectOption {
  label?: any
  value?: any
  [key: string]: any
}
```

and just add a lot of JS checks throughout! Including, perhaps, a validator on the prop :+1:

Reply:
Something to consider, and that can likely be a follow-up PR

Reply:
Spitballing here, I think the validator function could look something like:

```ts
	mounted() {
		const validOptions = this.options.every(
			option =>
				typeof option[this.labelField] === 'string' &&
				typeof option[this.valueField] === 'string'
		)
		if (!validOptions) console.warn(`
			The options provided do not all include the expected label & value string fields, and will not work as expected.
			Please either add 'label' and 'value' properties to each option, or use the 'labelField' and 'valueField'
			component properties to point to the display label & unique identifier properties on your options
		`)
	},
```